packages/grafana-data/src/datetime/rangeutil.ts

```diff
@@ -148,9 +148,17 @@ export function describeTextRange(expr: string): TimeOption {
  * ```
  * @category TimeUtils
  * @param range - a time range (usually specified by the TimePicker)
+ * @param timeZone - optional time zone.
+ * @param quickRanges - optional dashboard's custom quick ranges to pick range names from.
  * @alpha
  */
-export function describeTimeRange(range: RawTimeRange, timeZone?: TimeZone): string {
+export function describeTimeRange(range: RawTimeRange, timeZone?: TimeZone, quickRanges?: TimeOption[]): string {
+  for (const qr of quickRanges || []) {
+    if (range.from.toString() === qr.from && range.to.toString() === qr.to) {
+      return qr.display;
+    }
+  }
+
   const option = rangeIndex[range.from.toString() + ' to ' + range.to.toString()];
```

Comment:
Two requests on this:

We tend to prefer `arr.find` unless there's a reason not to. This seems like a perfect reason for it.

```ts
const foundOption = quickRanges.find(quickRange => range.from.toString() == quickRange.from && etc)
```

This is also basically the same check as below (`option = rangeIndex`), so I think they should be combined. Perhaps something like

```ts
const ranges = quickRanges ? rangeOptions.concat(quickRanges) : rangeOptions;
const foundOption = quickRanges.find(quickRange => range.from.toString() == quickRange.from && etc)
if (foundOption) {
  return foundOption.display;
}
```

I _would_ love to ask for tests for this new behaviour, but unfortunately it doesn't look like there's existing tests for this :(

Reply:
Reworked to use `.find`, added tests.